src/controllers/redeploy.go

```diff
@@ -4,13 +4,15 @@ import (
 	"github.com/devclub-iitd/DeployBot/src/history"
 )
 
-func redeploy(callbackID string, data map[string]interface{}) {
-	stop(callbackID, data)
-	state, _ := history.GetState(data["git_repo"].(string))
-	if state.Status == "stopped" {
-		data["subdomain"] = state.Subdomain
-		data["access"] = state.Access
-		data["server_name"] = state.Server
-		deploy(callbackID, data)
+func redeploy(params *DeployAction) {
+	state, _ := history.GetState(params.data["git_repo"].(string))
+
+	params.data["subdomain"] = state.Subdomain
+	params.data["access"] = state.Access
+	params.data["server_name"] = state.Server
+
+	if state.Status != "stopped" {
+		params.command = "redeploy"
 	}
+	handleDeploy(params)
```

Comment:
Instead of calling a deploy routine from redeploy.go, you should write an internal routine for redeploy. The routines are only slightly different right now but that might change in the future and making use of if statements can only complicate the flow. They can both call the same `deploy.sh` with different flags for now.